Source/WebCore/html/OffscreenCanvas.cpp

```diff
@@ -280,14 +297,21 @@ ExceptionOr<RefPtr<ImageBitmap>> OffscreenCanvas::transferToImageBitmap()
             return { ImageBitmap::create(ImageBitmapBacking(WTFMove(buffer))) };
         }
 
-        // As the canvas context state is stored in GraphicsContext, which is owned
-        // by buffer(), to avoid resetting the context state, we have to make a copy and
-        // clear the original buffer rather than returning the original buffer.
-        auto bufferCopy = buffer()->clone();
-        downcast<OffscreenCanvasRenderingContext2D>(*m_context).clearCanvas();
+        RefPtr<ImageBuffer> bitmap;
+        if (is<OffscreenCanvasRenderingContext2D>(*m_context)) {
+            // As the canvas context state is stored in GraphicsContext, which is owned
+            // by buffer(), to avoid resetting the context state, we have to make a copy and
+            // clear the original buffer rather than returning the original buffer.
+            bitmap = buffer()->clone();
+            downcast<OffscreenCanvasRenderingContext2D>(*m_context).clearCanvas();
+        } else {
+            bitmap = buffer();
+            m_hasCreatedImageBuffer = false;
+            setImageBuffer(nullptr);
```

Comment:
What's going on here? So if transferToImageBitmap is called on an ImageBitmapRenderingContext, it returns an empty image as well as destroying the buffer and clearing the context state? I'm probably just missing something because I don't recall the spec clearly, but this feels unintuitive and maybe warrants a comment, given the alternate code block and the comment there.

Reply:
The intention was that it just transfers the current buffer, with the effect that having no buffer is the same as having a blank buffer.

You make a good point about the context state (as the comment above states), so I probably should just do the same for both.